radial/core/radial_batch.py

```diff
@@ -157,3 +155,431 @@ def _assemble_load(self, results, *args, **kwargs):
+        return self
+
+    @action
+    @init_components
+    @inbatch_parallel(init="indices", target="threads")
+    def drop_outliers(self, index, contam=0.1, src=None, dst=None, **kwargs):
+        """Drop outliers using Isolation Forest algorithm.
+        Parameters
+        ----------
+        contam : float (from 0 to 0.5)
+            The amount of contamination of the data set
+        """
+        _ = kwargs
+        i = self.get_pos(None, src[0], index)
+
+        derivative = getattr(self, src[1])[i]
+        time = getattr(self, src[0])[i]
+
+        x_data = np.array([derivative]).T
+        isol = IsolationForest(contamination=contam).fit(x_data)
+        pred = isol.predict(x_data)
+        getattr(self, dst[0])[i] = time[pred == 1]
+        getattr(self, dst[1])[i] = derivative[pred == 1]
+        return self
+
+    @action
+    @init_components
+    @inbatch_parallel(init="indices", target="threads")
+    def get_samples(self, index, n_points, n_samples=1, sampler=None, # pylint: disable=too-many-arguments
+                    src=None, dst=None, interpolate='linear', seed=None, **kwargs):
+        """ Draws samples from the interpolation of time and derivative components.
+
+        Performs interpolation of the `derivative` on time using
+        `scipy.interpolate.interp1d`, samples `n_points` within `time` component
+        range, and calculates values of `derivative` in sampled points.
+        Sampler should return values from [0, 1] interval, which later is
+        stretched to the range on `time` component. Size of the sample equals
+        `(n_samples, n_points)`.
+
+        Parameters
+        ----------
+        n_points : int
+            Size of a sample.
+        n_samples : int, optional
+            Numper of samples. Defaults to `1`.
+        sampler : function or named expression
+            Method to sample points within [0, 1] range.
+            If callable, it should only take `size` as an argument.
+        interpolate : str, optional
+            Specifies the kind of interpolation as a string
+            ('linear', 'nearest', 'zero', 'slinear', 'quadratic', 'cubic',
+            'previous', 'next', where 'zero', 'slinear', 'quadratic' and 'cubic'
+            refer to a spline interpolation of zeroth, first, second or third
+            order; 'previous' and 'next' simply return the previous or next value
+            of the point) or as an integer specifying the order of the spline
+            interpolator to use.
+        seed : int, optional
+            Numpy seed to be fixed. Defaults to None.
+        kwargs : misc
+            Named arguments to the sampling
+            or interpolation functions.
+
+        Returns
+        -------
+        batch : RadialBatch
+            Batch with changed `time` and `derivative`
+            components. Changes components inplace.
+
+        Note
+        ----
+        Sampler should have following parameters: `tmin`, `tmax`, `size`. Other
```

Comment:
there are two ways of using sampler. If sampler is callable than it should only have ```size```, but if it name expression it can include another params, You right I'll add this in docs.